### PR TITLE
[dotnet] Refactor away private constructor from `Response`

### PR DESCRIPTION
**Pull request by RenderMichael**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

This removes one of the three constructors on the `Response` type, moving dictionary parsing responsibility to the correct method `FromJson`.

### Motivation and Context

Incremental step, part of desires I expressed in #14839 about making the `Response` type more immutable.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
enhancement

___

### **Description**
- Refactored the `Response` class to remove the private constructor that accepted a dictionary, simplifying the class design.
- Enhanced the `FromJson` method to handle JSON deserialization and object creation, improving code readability and maintainability.
- Improved error handling by throwing an exception when a null JSON response is encountered.
- Simplified the logic for setting `SessionId` and `Value` properties, ensuring more robust and clear code execution.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Response.cs</strong><dd><code>Refactor `Response` class to improve JSON handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/src/webdriver/Response.cs

<li>Removed the private constructor that accepted a dictionary.<br> <li> Refactored <code>FromJson</code> method to handle JSON deserialization and object <br>creation.<br> <li> Improved error handling for null JSON responses.<br> <li> Simplified logic for setting <code>SessionId</code> and <code>Value</code> properties.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14877/files#diff-c118ca207b54c33c677d0ea6b7d9379cb63ecfb92a537e512741f0b96013e77c">+22/-22</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: dotnet/src/webdriver/Response.cs [hunk 1]
```diff
@@ -56,57 +56,69 @@ public Response(SessionId sessionId)
...
+        /// Returns a new <see cref="Response"/> from a JSON-encoded string.
+        /// </summary>
+        /// <param name="responseJson">The JSON string to deserialize into a <see cref="Response"/>.</param>
+        /// <returns>A <see cref="Response"/> object described by the JSON string.</returns>
+        public static Response FromJson(string responseJson)
         {
+            Dictionary<string, object> rawResponse = JsonSerializer.Deserialize<Dictionary<string, object>>(responseJson, s_jsonSerializerOptions)
+                ?? throw new WebDriverException("JSON success response returned \"null\" value");
+
+            var @this = new Response();
```
Comment (nvborisenko) [inline dotnet/src/webdriver/Response.cs]: Variable name seems ugly, perhaps just `response` is better.

Comment (RenderMichael) [reply dotnet/src/webdriver/Response.cs]: I plan to propose some follow-up changes after this PR is merged, and this variable name makes the diffs as clean as possible. I can re-name in a follow-up, or I can change it here if that's preferred.

Comment (nvborisenko) [reply dotnet/src/webdriver/Response.cs]: Here and forget, please.

File: dotnet/src/webdriver/Response.cs [hunk 2]
```diff
@@ -56,57 +56,69 @@ public Response(SessionId sessionId)
             }
         }
 
-        private Response(Dictionary<string, object> rawResponse)
+        /// <summary>
+        /// Returns a new <see cref="Response"/> from a JSON-encoded string.
+        /// </summary>
+        /// <param name="responseJson">The JSON string to deserialize into a <see cref="Response"/>.</param>
+        /// <returns>A <see cref="Response"/> object described by the JSON string.</returns>
+        public static Response FromJson(string responseJson)
```
Comment (nvborisenko) [inline dotnet/src/webdriver/Response.cs]: Var name was `value`, just to avoid source breaking change.

Comment (RenderMichael) [reply dotnet/src/webdriver/Response.cs]: reverted

File: dotnet/src/webdriver/Response.cs (hunk 1 above)
Comment (RenderMichael) [reply dotnet/src/webdriver/Response.cs]: renamed to `response`

**Review by nvborisenko — APPROVED** [review]
Thanks @RenderMichael !

**Pull request merged**